### PR TITLE
Fixed NullPointerException with /col command in console.

### PR DESCRIPTION
**Pull request by Asherslab**
# Changes proposed in this pull request:
- Fixes the "NullPointerException" when executing the /col colony refresh ID (or the /col colony ownerchange ID PLAYER) command from console.
- Fixes /mc home between dimensions =D
- Fixes the time for /col colony info <colony> 
- also fixes delete command

Review please

### CODE REVIEW

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java [hunk 1]
```diff
@@ -6,6 +6,7 @@
 import com.minecolonies.coremod.commands.AbstractSingleCommand;
 import net.minecraft.command.CommandException;
 import net.minecraft.command.ICommandSender;
+import net.minecraft.entity.Entity;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java]: MINOR Remove this unused import 'net.minecraft.entity.Entity'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 27 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChangeColonyOwnerCommand.java#L118](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java#L118): A "NullPointerException" could be thrown; "colony" is nullable here [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS2259)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/90a78efef30a004ca3adb57ba85237afe0371c0e/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java [hunk 2]
```diff
@@ -53,7 +51,7 @@ public void execute(@NotNull final MinecraftServer server, @NotNull final IComma
     {
         if(args.length < 2)
         {
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java]: MAJOR Reduce the number of returns of this method 8, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 28 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 27 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [ChangeColonyOwnerCommand.java#L9](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java#L9): Remove this unused import 'net.minecraft.entity.Entity'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java (hunk 2 above)
Comment (Asherslab) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java]: That was already there.

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 27 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 26 minor

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ChangeColonyOwnerCommand.java#L50](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java#L50): Reduce the number of returns of this method 8, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/b6e8f62a93fa9aa9d1a376ce81d5d9b508600cca/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java [hunk 3]
```diff
@@ -51,9 +50,10 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
     @Override
     public void execute(@NotNull final MinecraftServer server, @NotNull final ICommandSender sender, @NotNull final String... args) throws CommandException
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/ChangeColonyOwnerCommand.java:51]: MAJOR Reduce the number of returns of this method 9, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/RefreshColonyCommand.java [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import com.mojang.authlib.GameProfile;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/RefreshColonyCommand.java]: MINOR Remove this unused import 'com.mojang.authlib.GameProfile'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/DeleteColonyCommand.java [hunk 5]
```diff
@@ -9,6 +9,7 @@
 import com.mojang.authlib.GameProfile;
```
Comment (MinecoloniesSonar) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/DeleteColonyCommand.java]: MINOR Remove this unused import 'com.mojang.authlib.GameProfile'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 30 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 29 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [DeleteColonyCommand.java#L22](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/commands/colonycommands/DeleteColonyCommand.java#L22): Remove this unused import 'java.util.UUID'. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AUselessImportCheck)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/d6552db229d75f7fb21b6a91219fa46c2f78af2d/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

File: src/main/java/com/minecolonies/coremod/commands/colonycommands/AddOfficerCommand.java [hunk 6]
```diff
@@ -50,19 +51,22 @@ public String getCommandUsage(@NotNull final ICommandSender sender)
...
-            sender.getCommandSenderEntity().addChatMessage(new TextComponentString(NO_ARGUMENTS));
+            sender.addChatMessage(new TextComponentString(NO_ARGUMENTS));
             return;
         }
 
+        final Entity senderEntity =  sender.getCommandSenderEntity();
+
         int colonyId = getIthArgument(args, 0, -1);
-        if(colonyId == -1 && sender instanceof EntityPlayer)
+        if(colonyId == -1 && sender instanceof EntityPlayer && senderEntity != null)
```
Comment (Raycoms) [inline src/main/java/com/minecolonies/coremod/commands/colonycommands/AddOfficerCommand.java]: can the senderEntity be null if the sender is instanceof EntityPlayer?

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 27 issues
* ![MAJOR][MAJOR] 1 major
* ![MINOR][MINOR] 26 minor

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] [blocks](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/blocks): Add a 'package-info.java' file to document the 'blocks' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [gui](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/client/gui): Add a 'package-info.java' file to document the 'gui' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [render](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/client/render): Add a 'package-info.java' file to document the 'render' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [colony](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/colony): Add a 'package-info.java' file to document the 'colony' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [buildings](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/colony/buildings): Add a 'package-info.java' file to document the 'buildings' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [permissions](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/colony/permissions): Add a 'package-info.java' file to document the 'permissions' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [creativetab](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/creativetab): Add a 'package-info.java' file to document the 'creativetab' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [entity](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/entity): Add a 'package-info.java' file to document the 'entity' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [baker](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/entity/ai/citizen/baker): Add a 'package-info.java' file to document the 'baker' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)
1. ![MINOR][MINOR] [farmer](https://github.com/Minecolonies/minecolonies/blob/892742e3b10073458b8209b2c3abf4fda096d5e5/src/main/java/com/minecolonies/coremod/entity/ai/citizen/farmer): Add a 'package-info.java' file to document the 'farmer' package [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1228)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by Raycoms** [comment]
:shipit:

[Approved with PullApprove](https://pullapprove.com/Minecolonies/minecolonies/pull-request/1252/?utm_source=github-pr&utm_medium=comment-badge&utm_campaign=Minecolonies/minecolonies)

**Comment by MinecoloniesSonar** [comment]
SonarQube analysis reported 44 issues
* ![MAJOR][MAJOR] 14 major
* ![MINOR][MINOR] 28 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [Colony.java](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java): This file has 1.282 lines, which is greater than 1.000 authorized. Split it into smaller files. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS00104)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): Split this class into smaller and more specialized ones to reduce its dependencies on other classes from 44 to the maximum authorized 30 or less. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1200)
1. ![MAJOR][MAJOR] [Colony.java#L51](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L51): class "Colony" has 86 methods, which is greater than the 35 authorized. Split it into smaller classes. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1448)
1. ![MAJOR][MAJOR] [Colony.java#L671](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L671): Reduce the number of returns of this method 11, down to the maximum allowed 6. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)
1. ![MAJOR][MAJOR] [Colony.java#L862](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L862): Refactor this method to reduce its Cognitive Complexity from 22 to the 20 allowed. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS3776)
1. ![MAJOR][MAJOR] [Colony.java#L1326](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1326): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1328](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1328): Assign this magic number 16 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1330](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1330): Assign this magic number 128 to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1449](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1449): Assign this magic number 0.5D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)
1. ![MAJOR][MAJOR] [Colony.java#L1449](https://github.com/Minecolonies/minecolonies/blob/8742677688fed69c0bd4a1c3102defe1c5a28da2/src/main/java/com/minecolonies/coremod/colony/Colony.java#L1449): Assign this magic number 0.1D to a well-named constant, and use the constant instead. [rule](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS109)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by marchermans — APPROVED** [review]

**Comment by Raycoms** [comment]
:shipit:

**Pull request merged**